cmd/v0.22/pingsource-cleanup/main.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/kelseyhightower/envconfig"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+
+	sourcesv1alpha2 "knative.dev/eventing/pkg/apis/sources/v1alpha2"
+)
+
+type envConfig struct {
+	SystemNamespace string `envconfig:"SYSTEM_NAMESPACE" default:"knative-eventing"`
+	DryRun          bool   `envconfig:"DRY_RUN" default:"false"`
+}
+
+func main() {
+	ctx := injectionEnabled()
+
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Fatalf("[ERROR] Failed to process env var: %s", err)
+	}
+
+	k8s := kubeclient.Get(ctx)
+	client := eventingclient.Get(ctx)
+
+	nss, err := k8s.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
```

Comment:
Can you update the RBAC rules in clusterroles.yaml to allow listing namespaces?

Reply:
done